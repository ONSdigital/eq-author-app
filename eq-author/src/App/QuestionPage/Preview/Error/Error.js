import PropTypes from "prop-types";
import styled, { css } from "styled-components";

import { colors } from "constants/theme";

const small = css`
  padding: 0.2em 0.5em;
  margin: 0;
  display: inline-block;
`;

const large = css`
  padding: 2em 1em;
`;

const Error = styled.div`
  padding: 0.5em;
  border: 2px dashed #b5c4cb;
  text-align: left;
  color: ${colors.secondary};
  font-size: 1em;
  font-weight: bold;
  margin-bottom: ${props => (props.margin ? "1em" : "0")};
  ${props => props.small && small};
  ${props => props.large && large};
`;

Error.defaultProps = {
  margin: true
};
Error.propTypes = {
  margin: PropTypes.bool,
  small: PropTypes.bool,
  large: PropTypes.bool
};

export default Error;
