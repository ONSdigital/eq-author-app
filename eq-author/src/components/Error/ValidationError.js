import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { colors } from "constants/theme";
import VisuallyHidden from "../VisuallyHidden/index";

import IconText from "components/IconText";
import WarningIcon from "constants/icon-warning.svg?inline";

const ErrorStyling = styled(IconText)`
  color: ${colors.red};
  width: ${(props) => (props.right ? "100%" : "80%")};
  justify-content: flex-end;
  margin: 0.5em 0;
`;

export const ValidationError = (props) => {
  const { children, right, className } = props;

  return (
    <ErrorStyling className={className} right={right} icon={WarningIcon}>
      <VisuallyHidden>Error:&nbsp;</VisuallyHidden>
      {children}
    </ErrorStyling>
  );
};

ValidationError.propTypes = {
  children: PropTypes.node.isRequired,
  right: PropTypes.bool,
  className: PropTypes.string,
};
