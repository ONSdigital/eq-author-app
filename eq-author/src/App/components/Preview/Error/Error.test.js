import React from "react";
import { shallow } from "enzyme";

import Error from "./Error";

describe("Error", () => {
  it("should render", () => {
    const wrapper = shallow(<Error />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render as large", () => {
    const wrapper = shallow(<Error large />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render as small", () => {
    const wrapper = shallow(<Error small />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with no margin", () => {
    const wrapper = shallow(<Error margin={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
