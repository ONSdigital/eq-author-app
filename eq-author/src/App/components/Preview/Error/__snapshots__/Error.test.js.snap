// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render 1`] = `
.c0 {
  padding: 0.5em;
  border: 2px dashed #b5c4cb;
  text-align: left;
  color: #5F7682;
  font-size: 1em;
  font-weight: bold;
  margin-bottom: 1em;
}

<div
  className="c0"
/>
`;

exports[`Error should render as large 1`] = `
.c0 {
  padding: 0.5em;
  border: 2px dashed #b5c4cb;
  text-align: left;
  color: #5F7682;
  font-size: 1em;
  font-weight: bold;
  margin-bottom: 1em;
  padding: 2em 1em;
}

<div
  className="c0"
/>
`;

exports[`Error should render as small 1`] = `
.c0 {
  padding: 0.5em;
  border: 2px dashed #b5c4cb;
  text-align: left;
  color: #5F7682;
  font-size: 1em;
  font-weight: bold;
  margin-bottom: 1em;
  padding: 0.2em 0.5em;
  margin: 0;
  display: inline-block;
}

<div
  className="c0"
/>
`;

exports[`Error should render with no margin 1`] = `
.c0 {
  padding: 0.5em;
  border: 2px dashed #b5c4cb;
  text-align: left;
  color: #5F7682;
  font-size: 1em;
  font-weight: bold;
  margin-bottom: 0;
}

<div
  className="c0"
/>
`;
